src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,293 @@
+    }
+    else
+    {
+        // Scattered off of nucleus
+        cos_theta = sample_cos_t(-1, rng);
+
+        // Calculate rejection for fake scattering
+        // TODO: Reference?
+        real_type mott_coeff
+            = 1 + real_type(1e-4) * ipow<2>(target_.atomic_number().get());
+        MottRatioCalculator mott_xsec(element_data_,
+                                      std::sqrt(particle_.beta_sq()));
+        real_type g_rej = mott_xsec(cos_theta)
+                          * ipow<2>(calculate_form_factor(cos_theta))
+                          / mott_coeff;
+
+        if (!BernoulliDistribution(g_rej)(rng))
+        {
+            return {0, 0, 1};
+        }
+    }
+
+    // Calculate scattered vector assuming azimuthal angle is isotropic
+    UniformRealDistribution<real_type> sample_phi(0, 2 * constants::pi);
+    return from_spherical(cos_theta, sample_phi(rng));
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the form factor based on the form factor model.
+ *
+ * The models are described in [LR15] section 2.4.2.1 and parameterize the
+ * charge distribution inside a nucleus. The same models are used as in
+ * Geant4, and are:
+ *      Exponential: [LR15] eqn 2.262
+ *      Gaussian: [LR15] eqn 2.264
+ *      Flat (uniform-uniform folded): [LR15] 2.265
+ */
+CELER_FUNCTION real_type
+WentzelDistribution::calculate_form_factor(real_type cos_t) const
+{
+    switch (data_.form_factor_type)
+    {
+        case NuclearFormFactorType::flat: {
+            real_type x1 = flat_coeff() * std::sqrt(mom_transfer_sq(cos_t));
+            real_type x0 = real_type{0.6} * x1
+                           * fastpow(value_as<Mass>(target_.nuclear_mass()),
+                                     real_type{1} / 3);
+            return flat_form_factor(x0) * flat_form_factor(x1);
+        }
+        case NuclearFormFactorType::exponential:
+            return 1
+                   / ipow<2>(
+                       1 + nuclear_form_prefactor() * mom_transfer_sq(cos_t));
+        case NuclearFormFactorType::gaussian:
+            return exp(-2 * nuclear_form_prefactor() * mom_transfer_sq(cos_t));
+        default:
+            return 1;
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Helper function for calculating the flat form factors, see [LR15] eqn
+ * 2.265.
+ */
+CELER_FUNCTION real_type WentzelDistribution::flat_form_factor(real_type x) const
+{
+    return 3 * (sin(x) - x * cos(x)) / ipow<3>(x);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Momentum coefficient used in the flat model for the nuclear form factors.
+ * This is the ratio of \f$ r_1 / \hbar \f$ where \f$ r_1 \f$ is defined in
+ * eqn 2.265 of [LR15].
+ */
+CELER_CONSTEXPR_FUNCTION real_type WentzelDistribution::flat_coeff() const
+{
+    return native_value_to<units::MevMomentum>(real_type(2e-15) * units::meter
```

Comment:
Feel free to add a `femtometer` to `celeritas/Units.hh` 😄 since I see that scale is used at least twice in this file so far